domain/ide/command/oauth_refresh_test.go

```diff
@@ -0,0 +1,112 @@
+	"github.com/stretchr/testify/assert"
+
+	"github.com/snyk/snyk-ls/application/config"
+	"github.com/snyk/snyk-ls/domain/observability/error_reporting"
+	"github.com/snyk/snyk-ls/domain/observability/ux"
+	"github.com/snyk/snyk-ls/domain/snyk"
+	"github.com/snyk/snyk-ls/infrastructure/cli/auth"
+	"github.com/snyk/snyk-ls/infrastructure/services"
+	"github.com/snyk/snyk-ls/infrastructure/snyk_api"
+	"github.com/snyk/snyk-ls/internal/lsp"
+	"github.com/snyk/snyk-ls/internal/notification"
+	"github.com/snyk/snyk-ls/internal/testutil"
+)
+
+func Test_oauthRefreshCommand_Execute_SameTokenNoUpdate(t *testing.T) {
+	testutil.UnitTest(t)
+	fakeApiClient := snyk_api.FakeApiClient{}
+	cmd := &oauthRefreshCommand{
+		command: snyk.CommandData{
+			CommandId: snyk.OAuthRefreshCommand,
+		},
+		authService: services.NewAuthenticationService(
+			&fakeApiClient,
+			&auth.FakeAuthenticationProvider{IsAuthenticated: true},
+			ux.NewTestAnalytics(),
+			error_reporting.NewTestErrorReporter(),
+		),
+	}
+
+	c := config.CurrentConfig()
+	c.SetAuthenticationMethod(lsp.OAuthAuthentication)
+	ctrl := gomock.NewController(t)
+	mockEngine := mocks.NewMockEngine(ctrl)
+	mockEngine.EXPECT().GetConfiguration().Return(c.Engine().GetConfiguration()).AnyTimes()
+	mockEngine.EXPECT().Invoke(localworkflows.WORKFLOWID_WHOAMI)
+	c.SetEngine(mockEngine)
+
+	err := cmd.Execute(context.Background())
+
+	assert.NoErrorf(t, err, "cmd.Execute() error = %v", err)
+}
+
+func Test_oauthRefreshCommand_Execute_DifferentTokenUpdate(t *testing.T) {
+	testutil.UnitTest(t)
+	fakeApiClient := snyk_api.FakeApiClient{}
+	analytics := ux.NewTestAnalytics()
+	cmd := &oauthRefreshCommand{
+		command: snyk.CommandData{
+			CommandId: snyk.OAuthRefreshCommand,
+		},
+		authService: services.NewAuthenticationService(
+			&fakeApiClient,
+			&auth.FakeAuthenticationProvider{IsAuthenticated: true},
+			analytics,
+			error_reporting.NewTestErrorReporter(),
+		),
+	}
+
+	c := config.CurrentConfig()
+	c.SetAuthenticationMethod(lsp.OAuthAuthentication)
+	ctrl := gomock.NewController(t)
+	mockEngine := mocks.NewMockEngine(ctrl)
+
+	engineConfig := c.Engine().GetConfiguration()
+	c.SetEngine(mockEngine)
+	mockEngine.EXPECT().GetConfiguration().Return(engineConfig).AnyTimes()
+	mockEngine.EXPECT().Invoke(localworkflows.WORKFLOWID_WHOAMI)
+	engineConfig.Set(auth2.CONFIG_KEY_OAUTH_TOKEN, "something different")
+	assert.NotEqual(t, c.Token(), engineConfig.GetString(auth2.CONFIG_KEY_OAUTH_TOKEN), "token should be different")
+
+	notification.DisposeListener()
+	receivedChan := make(chan bool)
+	notification.CreateListener(func(params any) {
+		if reflect.TypeOf(params) == reflect.TypeOf(lsp.AuthenticationParams{}) {
+			receivedChan <- true
+		}
+	})
+
+	err := cmd.Execute(context.Background())
+
+	assert.NoErrorf(t, err, "cmd.Execute() error = %v", err)
+	assert.Equal(t, c.Token(), engineConfig.GetString(auth2.CONFIG_KEY_OAUTH_TOKEN))
+	assert.True(t, analytics.Identified)
+	assert.True(t, <-receivedChan)
```

Comment:
This will block test runs and time out in 10 mins only, if the listener won't hit.

Reply:
can use assert.Eventually instead